addons/resource/models/resource_calendar.py

```diff
@@ -108,27 +109,147 @@ def default_get(self, fields):
+    def _compute_two_weeks_attendance(self):
+        for calendar in self:
+            if not calendar.two_weeks_calendar:
+                continue
+            calendar.attendance_ids_1st_week = calendar.attendance_ids.filtered(lambda a: a.week_type == '0')
+            calendar.attendance_ids_2nd_week = calendar.attendance_ids.filtered(lambda a: a.week_type == '1')
+
+    def _inverse_two_weeks_calendar(self):
+        for calendar in self:
+            if not calendar.two_weeks_calendar:
+                continue
+            calendar.attendance_ids = calendar.attendance_ids_1st_week + calendar.attendance_ids_2nd_week
+
+    @api.depends("schedule_type")
+    def _compute_flexible_hours(self):
+        for calendar in self:
+            calendar.flexible_hours = calendar.schedule_type in ('fully_flexible', 'flexible')
+
+    def _inverse_flexible_hours(self):
+        for calendar in self:
+            if not calendar.schedule_type:
+                calendar.schedule_type = 'flexible' if calendar.flexible_hours else 'fully_fixed'
+
+    @api.depends('attendance_ids', 'attendance_ids.hour_from', 'attendance_ids.hour_to', 'two_weeks_calendar', 'schedule_type', 'fixed_time_with_hours',
+                 'monday', 'tuesday', 'wednesday', 'thursday', 'friday', 'saturday', 'sunday',
+                 'monday_hours', 'tuesday_hours', 'wednesday_hours', 'thursday_hours', 'friday_hours', 'saturday_hours', 'sunday_hours',
+                 'monday_2', 'tuesday_2', 'wednesday_2', 'thursday_2', 'friday_2', 'saturday_2', 'sunday_2',
+                 'monday_hours_2', 'tuesday_hours_2', 'wednesday_hours_2', 'thursday_hours_2', 'friday_hours_2', 'saturday_hours_2', 'sunday_hours_2')
     def _compute_hours_per_day(self):
         for calendar in self:
             if calendar.flexible_hours:
                 continue
-            attendances = calendar._get_global_attendances()
-            calendar.hours_per_day = calendar._get_hours_per_day(attendances)
 
-    @api.depends('attendance_ids.hour_from', 'attendance_ids.hour_to')
+            if calendar.schedule_type == 'fully_fixed':
+                attendances = calendar._get_global_attendances()
+                calendar.hours_per_day = calendar._get_hours_per_day(attendances)
+
+            elif calendar.schedule_type == 'fixed_time':
+                if not calendar.fixed_time_with_hours:
+                    continue
+
+                cnt, total = 0, 0
+                for day in ['monday', 'tuesday', 'wednesday', 'thursday', 'friday', 'saturday', 'sunday']:
+                    if calendar[day]:
+                        cnt += 1
+                        total += calendar[f'{day}_hours']
+                    if calendar[f'{day}_2']:
+                        cnt += 1
+                        total += calendar[f'{day}_hours_2']
+                if not cnt:
+                    return 0
+
+                calendar.hours_per_day = total / cnt
+
+    def _compute_hours_per_week_value(self):
```

Comment:
This method isn't a compute and shouldn't be named as such :slightly_smiling_face: